### PR TITLE
Order bulk create allow to import archive orders

### PR DESCRIPTION
**Pull request by kadewu**
Fix: https://github.com/saleor/saleor/issues/11720

To enable importing archive orders, we for now we introduce 2 changes to bulk create:
- allow to not provide user at all
- allow to skip variants

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation: https://github.com/saleor/saleor-docs/pull/968

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -78,6 +78,11 @@ All notable, unreleased changes to this project will be documented in this file.
 
 ### Breaking changes
 
+- **Feature preview breaking change**:
+  - `orderBulkCreate` now will attempt to create order with `IGNORE_FAILED` policy even if: - #14177 by @kadewu
+    - `User` cannot be resolved, and `email` wasn't provided.
+    - `Variant` wasn't provided.
```
Comment (IKarbowiak) [inline CHANGELOG.md:84]:
```suggestion
  - Order bulk create allow to import archive orders - #14177 by @kadewu
    - `orderBulkCreate` now will attempt to create order with `IGNORE_FAILED` policy even if:
      - `User` cannot be resolved, and `email` wasn't provided.
      - `Variant` wasn't provided.
```

Comment (IKarbowiak) [reply CHANGELOG.md:84]: I'm also wondering if those changes are really breaking? 🤔 Like we are not cutting the possibility, but extending it

File: saleor/graphql/order/bulk_mutations/order_bulk_create.py [hunk 2]
```diff
@@ -1700,7 +1709,13 @@ def create_single_fulfillment(
...
                 return None
 
-            if order_line.line.variant.id != variant.id:
+            if (
+                variant
+                and order_line.line.variant
+                and order_line.line.variant.id != variant.id
+                or (variant and not order_line.line.variant)
+                or (not variant and order_line.line.variant)
+            ):
```
Comment (IKarbowiak) [inline saleor/graphql/order/bulk_mutations/order_bulk_create.py:1718]: Could we split this into smaller part. I'm thinking about introducing variables to dedicated check, like: `line_variant_mismatch` and `missing_line_variant` and then use it here. What do you think?

File: CHANGELOG.md (hunk 1 above)
Comment (kadewu) [reply CHANGELOG.md:84]: Actually, I also wondered how to address that change, I would like to make sure, that clients that are using already that mutations are aware, that they now create some orders when using said policy. But I'm fine with changing where we include that information.

File: saleor/graphql/order/bulk_mutations/order_bulk_create.py (hunk 2 above)
Comment (kadewu) [reply saleor/graphql/order/bulk_mutations/order_bulk_create.py:1718]: Sure, it makes sense.

File: CHANGELOG.md (hunk 1 above)
Comment (IKarbowiak) [reply CHANGELOG.md:84]: @maarcingebala what do you think?

**Review by zedzior — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Comment by JannikZed** [comment]
Just for clarification: we can now set variant and product name and if the product/variants do not exist yet in Saleor, Saleor will just store this information and will provide it later when querying the order? Saleor variantId will be empty in this case?

**Pull request merged**